File: video_1.1_data_import/data_import.R
# import my dataset

mycsv <- "Russian_Miscanthus.csv"

mydata <- read.csv(mycsv, stringsAsFactors = FALSE)

#1+1 # demo of Ctrl+Enter

str(mydata) # examine data structure

# exercise with a dummy dataset
testdata <- read.csv("testcsv.csv",
                     stringsAsFactors = FALSE)
str(testdata)

# extracting individual columns
mydata$Species # individual columns are vectors
is.vector(mydata$Species)
is.vector(mycsv)

mydata$Coll..No..Accession.ID[15]
mydata$Species[15]
mydata$Lat[15]
mydata$Long[15]

mydata[15,]

# creating a vector
testvect <- c(2, 3, 4)
testvect

# fixing column names
colnames(mydata)
colnames(mydata) <- c("Accession", "Species",
                      "Latitude", "Longitude",
                      "Elevation", "Stem_height",
                      "Stem_diameter", "Stem_density")

mydata$Accession[15]
mydata$Species[15]
mydata$Lat[15]
mydata$Long[15]

# doing arithmetic and adding columns
mydata$Stem_height_cm <- mydata$Stem_height * 100

mydata$Elevation_ft <- mydata$Elevation * 3.28

str(mydata)

File: video_1.1_data_import/data_import.py
import numpy
import csv

mycsv = "Russian_Miscanthus.csv"

mydata = numpy.genfromtxt(mycsv, delimiter = ',', skip_header = 1,
                 usecols = (2, 3, 4, 5, 6, 7))

#print(mydata)

# this is a comment!
# assign columns of mydata to meaningful variable names
latitude = mydata[:,0]
longitude = mydata[:,1]
elevation = mydata[:,2]
stem_height = mydata[:,3]
stem_diameter = mydata[:,4]
stem_density = mydata[:,5]

# importing text columns
accession = []
species = []
with open(mycsv, newline = '', mode = 'r') as mycon:
    myreader = csv.reader(mycon)
    header = next(myreader)
    for row in myreader:
        accession.append(row[0])
        species.append(row[1])
        
print(accession[14])
print(species[14])
print(latitude[14])
print(longitude[14])

stem_height_cm = stem_height * 100
print(stem_height[:20])    # print out first 20 to check
print(stem_height_cm[:20])

elevation_ft = elevation * 3.28
print(elevation[:20])    # print out first 20 to check
print(elevation_ft[:20])
